torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -716,7 +783,11 @@ std::shared_ptr<JitFuture> TensorPipeAgent::send(
         pipeRead(
             clientPipe.pipe_,
             [this, &clientPipe](
-                const tensorpipe::Error& error, Message&& responseMessage) {
+                const tensorpipe::Error& error,
+                Message&& responseMessage,
+                // NOLINTNEXTLINE(performance-unnecessary-value-param)
+                std::shared_ptr<FullDeviceContext> ctx) {
+              ctx->blockCurrentStreams();
```

Comment:
I don't understand why we're making the current streams wait for the context. If I remember correctly, this callback could be invoked while inside a TensorPipe event loop thread, hence the current streams of that thread might be meaningless.

Moreover, the synchronization of the return value of an RPC call should be taken care of entirely by the CUDAFuture class, without any extra steps by the agent.